packages/next/src/client/components/redirect.ts

```diff
@@ -37,18 +37,21 @@ export function getRedirectError(
  *
  * - In a Server Component, this will insert a meta tag to redirect the user to the target page.
  * - In a Route Handler or Server Action, it will serve a 307/303 to the caller.
+ * - In a Server Action, type defaults to 'push' and 'replace' elsewhere.
  *
  * Read more: [Next.js Docs: `redirect`](https://nextjs.org/docs/app/api-reference/functions/redirect)
  */
 export function redirect(
   /** The URL to redirect to */
   url: string,
-  type: RedirectType = RedirectType.replace
+  type?: RedirectType
 ): never {
   const actionStore = actionAsyncStorage.getStore()
+  const redirectType =
+    type || (actionStore?.isAction ? RedirectType.push : RedirectType.replace)
```

Comment:
I'm not following why this change was needed if we're honoring the redirectType that is used by the user

Reply:
if a `type` is provided, we use it.. if not, then we have to default to `push` when redirect happens inside an action, and `replace` otherwise. From the docs: https://arc.net/l/quote/uockedeg

Reply:
To elaborate:

Earlier, the `type` argument had a default value of `RedirectType.replace`. This meant that in a server action, if we call `redirect('url')`, the `type` value would be set to `replace` by default, whereas it should be `push` as per the docs:
<img width="691" alt="image" src="https://github.com/user-attachments/assets/102867b5-ea19-4224-a25d-e70632dae7f0">

In order to set the default value of `type`, we need to know if we are running inside an action or not. That is why this change is made.